### PR TITLE
Add a generic function compiler to handle `IGNORE NULLS` in window functions

### PR DESCRIPTION
**Pull request by laserkaplan**
## Description
Adds a simple compiler method for using the `IGNORE NULLS` functionality of window functions in Trino. I implemented this privately in a package for work but figured I could generalize it and add it to the package as a whole :)

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: tests/unit/sqlalchemy/test_compiler.py [hunk 1]
```diff
@@ -113,3 +113,30 @@ def test_table_clause(dialect):
...
+        ('nth_value', func.nth_value),
+        ('lead', func.lead),
+        ('lag', func.lag),
+    ]
+)
+def test_ignore_nulls(dialect, function, element):
+    statement = select(
+        element(
+            table_without_catalog.c.id,
+            ignore_nulls=True,
```
Comment (hovaesco) [inline tests/unit/sqlalchemy/test_compiler.py:136]: add a test with `ignore_nulls=False`

**Review by hovaesco — APPROVED** [review]

**Comment by hovaesco** [comment]
Please squash commits and rebase on master.

**Comment by laserkaplan** [comment]
> Please squash commits and rebase on master.

Took a couple of tries but done now :P

**Review by hovaesco — APPROVED** [review]

**Comment by hashhar** [comment]
just reworded the commit message for release note's sake. Merging.

**Pull request merged**